packages/eslint-plugin/src/rules/prefer-optional-chain.ts

```diff
@@ -109,7 +116,87 @@ export default createRule<
     const seenLogicals = new Set<TSESTree.LogicalExpression>();
 
     return {
-      // specific handling for `(foo ?? {}).bar` / `(foo || {}).bar`
+      // specific handling for `if (foo) { foo.bar(); }` / `if (foo) foo.bar();`
+      'IfStatement[consequent.type=BlockStatement][consequent.body.length=1], IfStatement[consequent.type=ExpressionStatement]'(
+        node: TSESTree.IfStatement,
+      ): void {
+        if (!options.allowIfStatements || node.alternate) {
+          return;
+        }
+        const ifBodyStatement =
+          node.consequent.type === AST_NODE_TYPES.BlockStatement
+            ? node.consequent.body[0]
+            : node.consequent;
+
+        if (ifBodyStatement.type !== AST_NODE_TYPES.ExpressionStatement) {
+          return;
+        }
+
+        const ifBodyExpression = ifBodyStatement.expression;
+        const currentChain: ValidOperand[] = [
+          {
+            node: node.test,
+            type: OperandValidity.Valid,
+            comparedName: node.test,
+            comparisonType: NullishComparisonType.Boolean,
+            isYoda: false,
+          },
+        ];
+        if (node.test.type === AST_NODE_TYPES.LogicalExpression) {
+          const { newlySeenLogicals, operands } = gatherLogicalOperands(
+            node.test,
+            parserServices,
+            context.sourceCode,
+            options,
+          );
+          for (const logical of newlySeenLogicals) {
+            seenLogicals.add(logical);
+          }
+          for (const operand of operands) {
+            if (operand.type === OperandValidity.Invalid) {
+              return;
+            }
+            currentChain.push(operand);
+          }
+        }
+        if (ifBodyExpression.type === AST_NODE_TYPES.LogicalExpression) {
+          const { newlySeenLogicals, operands } = gatherLogicalOperands(
+            ifBodyExpression,
+            parserServices,
+            context.sourceCode,
+            options,
+          );
+          for (const logical of newlySeenLogicals) {
+            seenLogicals.add(logical);
+          }
+          for (const operand of operands) {
+            if (operand.type === OperandValidity.Invalid) {
+              return;
+            }
+            currentChain.push(operand);
+          }
+        } else {
+          if (ifBodyExpression.type !== AST_NODE_TYPES.CallExpression) {
+            return;
+          }
```

Comment:
[Bug] That's not for this rule to say. Ignoring unused property accesses is in the realm of [`@typescript-eslint/no-unused-expressions`](https://typescript-eslint.io/rules/no-unused-expressions). Folks should be able get this optional chaining complaint on code like:

```ts
if (a) {
  a.b;
}
```

There are codebases that use getters for control flow. I hate it, almost everyone hates it, but it's a legitimate choice that this rule shouldn't judge.  For example, `.to.be.*;` in Chai: https://www.chaijs.com/api/bdd/#method_null / https://www.chaijs.com/api/bdd/#extensible / etc.


```suggestion
```